internal/evaluator/filters.go

```diff
@@ -353,3 +390,1014 @@ func extractStringArrayFromRuleData(src ecc.Source, key string) []string {
+// (same logic as ECPolicyResolver).
+func (r *IncludeExcludePolicyResolver) determinePackageInclusion(pkg string, pkgRules []rule.Info, target string, result *PolicyResolutionResult) {
+	r.baseDeterminePackageInclusion(pkg, pkgRules, target, result)
+}
+
+// Includes returns the include criteria used by this policy resolver
+func (r *IncludeExcludePolicyResolver) Includes() *Criteria {
+	return r.include
+}
+
+// Excludes returns the exclude criteria used by this policy resolver
+func (r *IncludeExcludePolicyResolver) Excludes() *Criteria {
+	return r.exclude
+}
+
+// baseEvaluateRuleInclusion contains the shared logic for evaluating rule inclusion
+func (r *basePolicyResolver) baseEvaluateRuleInclusion(ruleID string, ruleInfo rule.Info, target string, result *PolicyResolutionResult) {
+	// Create matchers for this rule (similar to makeMatchers in conftest_evaluator.go)
+	matchers := r.createRuleMatchers(ruleID, ruleInfo)
+
+	// Score against include criteria
+	includeScore := r.scoreMatches(matchers, r.include.get(target), result.MissingIncludes)
+
+	// Score against exclude criteria
+	excludeScore := r.scoreMatches(matchers, r.exclude.get(target), make(map[string]bool))
+
+	// Debug: Log rule scoring
+	log.Debugf("[evaluateRuleInclusion] Rule: %s, includeScore: %d, excludeScore: %d, matchers: %v", ruleID, includeScore, excludeScore, matchers)
+
+	// Determine inclusion based on scores
+	if includeScore > excludeScore {
+		result.IncludedRules[ruleID] = true
+		result.Explanations[ruleID] = fmt.Sprintf("included (include score: %d, exclude score: %d)", includeScore, excludeScore)
+		log.Debugf("[evaluateRuleInclusion] Rule: %s INCLUDED", ruleID)
+	} else if excludeScore > 0 {
+		result.ExcludedRules[ruleID] = true
+		result.Explanations[ruleID] = fmt.Sprintf("excluded (include score: %d, exclude score: %d)", includeScore, excludeScore)
+		log.Debugf("[evaluateRuleInclusion] Rule: %s EXCLUDED", ruleID)
+	} else {
+		// No explicit criteria, check default behavior
+		if len(r.include.get(target)) == 0 || (len(r.include.get(target)) == 1 && r.include.get(target)[0] == "*") {
+			result.IncludedRules[ruleID] = true
+			result.Explanations[ruleID] = "included by default (no explicit includes)"
+			log.Debugf("[evaluateRuleInclusion] Rule: %s INCLUDED by default", ruleID)
+		} else {
+			result.Explanations[ruleID] = "not explicitly included"
+			log.Debugf("[evaluateRuleInclusion] Rule: %s NOT explicitly included", ruleID)
+		}
+	}
+}
+
+// baseDeterminePackageInclusion contains the shared logic for determining package inclusion
+func (r *basePolicyResolver) baseDeterminePackageInclusion(pkg string, pkgRules []rule.Info, target string, result *PolicyResolutionResult) {
+	// Check if any rule in the package is included
+	hasIncludedRules := false
+	hasExcludedRules := false
+
+	for _, ruleInfo := range pkgRules {
+		ruleID := ruleInfo.Code
+		if result.IncludedRules[ruleID] {
+			hasIncludedRules = true
```

Comment:
given the `if-else` block below, you can add a break here
```suggestion
			hasIncludedRules = true
			break
```

Reply:
idk. If it has both hasInclude and hasExclude, it's considered included.